spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/StackdriverJsonLayout.java

```diff
@@ -173,7 +173,7 @@ public void addLoggingEventEnhancer(String enhancerClassName) {
 			loggingEventEnhancers.add(clz.getDeclaredConstructor().newInstance());
 		}
 		catch (Exception ex) {
-			throw new RuntimeException(ex);
+			throw new EnhancedLoggingEventException(ex);
```

Comment:
I see. I wasn't sure what Sonar's reason for the exception subtype was -- I imagined it is just so the user has more context right?

What do you think of just adding a message for the user instead to add context:

```
throw new RuntimeException("Failed to a StackdriverJsonLayout logging enhancer class: " + enhancerClassName, ex);
```

This might achieve the intended effect without needing the extra subclass.

Reply:
SonarCloud's reasoning below:

```
Using such generic exceptions as Error, RuntimeException, Throwable, and Exception prevents calling methods from handling true, system-generated exceptions differently than application-generated errors.
```

So I think SC would still raise the same objection because you're still throwing a generic `RuntimeException` directly.

Reply:
I kind of agree with Sonar here.

Reply:
Ahh, I see. Ok let's just introduce the new `public` exception class then.

Reply:
I vote in favor of introducing new custom `RuntimeException` subtypes as normal practice. They make it easier to debug at a glance, and yes, allow special handling by downstream code. Making users rely on exception message parsing turns literal strings into a set-in-stone part of the API, which sucks.

Reply:
Sorry for getting hung up on this, but I just looked at this method more carefully, and the only reason we would get an exception here is when an invalid class name is passed in. Either the class is not found, or it doesn't have an appropriate constructor. So, I think it might make a lot of sense to just throw an IllegalArgumentException with an additional message that the class that was passed in is invalid.